packages/file/lib/src/backends/memory/node.dart

```diff
@@ -46,6 +47,10 @@ abstract class NodeBasedFileSystem implements StyleableFileSystem {
   /// The path of the current working directory.
   String get cwd;
 
+  /// The clock to use when finding the current time (e.g. to set the creation
+  /// time of a new node).
+  MemoryFileSystemClock get clock;
```

Comment:
`NodeBasedFileSystem` is layered beneath `MemoryFileSystem`, so we shouldn't refer to a "memory file system clock" here.

Since the interface of the clock has nothing to do with memory file system, what about naming it `Clock`, of if you're worried about naming collisions, maybe `SimpleClock`, `TimeProvider`, ...?

Reply:
Used `Clock`.